### PR TITLE
Show link dialog on long press

### PR DESCRIPTION
**Pull request by pipe01**
When long pressing a post thumbnail or image a dialog will show containing actions related to the post's link, similar to Boost. If the link is an image, an option to download it will be added.

image

### CODE REVIEW

File: app/src/main/java/com/jerboa/Utils.kt [hunk 1]
```diff
@@ -881,6 +884,23 @@ fun saveBitmap(
...
+    val extension = MimeTypeMap.getFileExtensionFromUrl(url)
+    val mimeType = MimeTypeMap.getSingleton().getMimeTypeFromExtension(extension)
+
+    withContext(Dispatchers.IO) {
+        URL(url).openStream().use {
+            saveBitmap(context, it, mimeType, fileName)
+        }
+    }
+
+    Toast.makeText(context, "Saved image", Toast.LENGTH_SHORT).show()
```
Comment (MV-GH) [inline app/src/main/java/com/jerboa/Utils.kt:901]: "Saved image" should be a stringResource

**Review by twizmwazin — CHANGES_REQUESTED** [review]
Please use string resources over literals, then it'll lgtm.

File: app/src/main/java/com/jerboa/Utils.kt [hunk 2]
```diff
@@ -881,6 +884,23 @@ fun saveBitmap(
     }
 }
 
+suspend fun saveImage(url: String, context: Context) {
+    Toast.makeText(context, "Saving image...", Toast.LENGTH_SHORT).show()
```
Comment (dessalines) [inline app/src/main/java/com/jerboa/Utils.kt:888]: stringResource.

**Comment by MV-GH** [comment]
Obsolete by #1189

**Pull request closed without merging**